.circleci/config.yml

```diff
@@ -11,11 +11,17 @@ commands:
       test_params:
         type: string
         default: ""
-    steps: 
+    steps:
       - checkout
       - run:
           name: Submodule checkout
-          command: git submodule update --init --recursive
+          command: |
```

Comment:
@rafie can you explain what is this for?

Reply:
This is an optimization for the dependency checkout. If we perform coverage analysis following the controlling build, we can save the dependencies in the build and retrieve them in the coverage phase. However, it looks that I might revert it if it turns out that the time gain doesn't worth the complexity.

Reply:
:+1: